src/Microsoft.Health.Fhir.Shared.Core/Features/Search/Behaviors/ListSearchPipeBehavior.cs

```diff
@@ -0,0 +1,131 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using Hl7.Fhir.Model;
+using MediatR;
+using Microsoft.Health.Extensions.DependencyInjection;
+using Microsoft.Health.Fhir.Core.Extensions;
+using Microsoft.Health.Fhir.Core.Features.Persistence;
+using Microsoft.Health.Fhir.Core.Features.Search.SearchValues;
+using Microsoft.Health.Fhir.Core.Messages.Search;
+using Microsoft.Health.Fhir.Core.Models;
+
+namespace Microsoft.Health.Fhir.Core.Features.Search.Behavior
+{
+    /// <summary>
+    /// This behavior looks for searches that use _list then resolves the list and rewrites the query with the resource _ids
+    /// </summary>
+    public class ListSearchPipeBehavior : IPipelineBehavior<SearchResourceRequest, SearchResourceResponse>
+    {
+        private const string _listParameter = "_list";
+        private const string _idParameter = "_id";
+        private readonly IScoped<IFhirDataStore> _dataStore;
+        private readonly ResourceDeserializer _deserializer;
+        private readonly IBundleFactory _bundleFactory;
+        private readonly IReferenceSearchValueParser _referenceSearchValueParser;
+        private readonly ISearchOptionsFactory _searchOptionsFactory;
+
+        public ListSearchPipeBehavior(
+            ISearchOptionsFactory searchOptionsFactory,
+            IBundleFactory bundleFactory,
+            IScoped<IFhirDataStore> dataStore,
+            ResourceDeserializer deserializer,
+            IReferenceSearchValueParser referenceSearchValueParser)
+        {
+            EnsureArg.IsNotNull(bundleFactory, nameof(bundleFactory));
+            EnsureArg.IsNotNull(dataStore, nameof(dataStore));
+            EnsureArg.IsNotNull(deserializer, nameof(deserializer));
+            EnsureArg.IsNotNull(referenceSearchValueParser, nameof(referenceSearchValueParser));
+            EnsureArg.IsNotNull(searchOptionsFactory, nameof(searchOptionsFactory));
+
+            _searchOptionsFactory = searchOptionsFactory;
+            _bundleFactory = bundleFactory;
+            _dataStore = dataStore;
+            _deserializer = deserializer;
+            _referenceSearchValueParser = referenceSearchValueParser;
+        }
+
+        public async Task<SearchResourceResponse> Handle(SearchResourceRequest request, CancellationToken cancellationToken, RequestHandlerDelegate<SearchResourceResponse> next)
+        {
+            EnsureArg.IsNotNull(request, nameof(request));
+
+            Tuple<string, string> listParameter = request.Queries
+                .FirstOrDefault(x => string.Equals(x.Item1, _listParameter, StringComparison.Ordinal));
```

Comment:
Is this case-sensitive on purpose?
I see cases where other search parameters are compared without case, like [this](https://github.com/microsoft/fhir-server/blob/6e1ed88eb20a9b73a67399870963d12ed1f786b4/src/Microsoft.Health.Fhir.Core/Extensions/SearchServiceExtensions.cs#L28).

Reply:
good catch.